h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -1542,6 +1548,54 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             }
             break;
         }
+        case ARRAY_CONCAT: {
+            final ValueArray array = (ValueArray) v0.convertTo(Value.ARRAY);
+            final ValueArray array2 = (ValueArray) v1.convertTo(Value.ARRAY);
+            if (!array.getComponentType().equals(array2.getComponentType()))
+                throw DbException.get(ErrorCode.GENERAL_ERROR_1, "Expected component type " + array.getComponentType()
```

Comment:
I don't think that `GENERAL_ERROR_1` is a good choice here, but because H2 can't store component type it does not matter now.